HomeWork18/HomeWork18/Loggers/Logger.cs

```
@@ -0,0 +1,102 @@
+            {
+                OnBackupRequired();
+
+            }
+
+            await WriteToFileAsync(message);
+        }
+
+        private async Task WriteToFileAsync(string message)
+        {
+            try
+            {
+                using (StreamWriter write = File.AppendText(_logFilePath))
+                {
+                    await write.WriteLineAsync(message).ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error writing to file: {ex.Message}");
+
+            }
+        }
+
+        protected virtual void OnBackupRequired()
+        {
+            lock (_lockObject)
+            {
+                BackupEventHandler handler = BackupRequired;
+                if (handler != null)
+                {
+                    handler.Invoke();
+                }
+            }
+        }
+
+        public async Task LoadConfigAsync()
+        {
+            try
+            {
+                string configFilePath = "config.json";
+                if (File.Exists(configFilePath))
+                {
+                    string json = await File.ReadAllTextAsync(configFilePath).ConfigureAwait(false);
+                    var config = JsonSerializer.Deserialize<Config>(json);
+
+                    backupThreshold = config.BackupThreshold;
+                }
+                else
+                {
+                    var config = new Config { BackupThreshold = 10 };
+                    var options = new JsonSerializerOptions { WriteIndented = true };
+                    var jsonString = JsonSerializer.Serialize(config, options);
+                    await File.WriteAllTextAsync(configFilePath, jsonString).ConfigureAwait(false);
+
+                    backupThreshold = config.BackupThreshold;
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error loading configuration: {ex.Message}");
```

Better to throw an exception